src/back/msl/keywords.rs

```diff
@@ -0,0 +1,100 @@
+//TODO: find a complete list
+pub const RESERVED: &[&str] = &[
+    // control flow
+    "break",
+    "if",
+    "else",
+    "continue",
+    "goto",
+    "do",
+    "while",
+    "for",
+    "switch",
+    "case",
+    // types and values
+    "void",
+    "unsigned",
+    "signed",
+    "bool",
+    "char",
+    "int",
+    "long",
+    "float",
+    "double",
+    "char8_t",
+    "wchar_t",
+    "true",
+    "false",
+    "nullptr",
+    "union",
+    "class",
+    "struct",
+    "enum",
+    // other
+    "main",
+    "decltype",
+    "sizeof",
+    "typeof",
+    "typedef",
+    "explicit",
+    "export",
+    "friend",
+    "namespace",
+    "operator",
+    "public",
+    "template",
+    "typename",
+    "typeid",
+    "co_await",
+    "co_return",
+    "co_yield",
+    "module",
+    "import",
+    "ray_data",
+    "vec_step",
+    "visible",
+    "as_type",
+    // qualifiers
+    "mutable",
+    "static",
+    "volatile",
+    "restrict",
+    "const",
+    "non-temporal",
+    "dereferenceable",
+    "invariant",
+    // exceptions
+    "throw",
+    "try",
+    "catch",
+    // operators
+    "const_cast",
+    "dynamic_cast",
+    "reinterpret_cast",
+    "static_cast",
+    "new",
+    "delete",
+    "and",
+    "and_eq",
+    "bitand",
+    "bitor",
+    "compl",
+    "not",
+    "not_eq",
+    "or",
+    "or_eq",
+    "xor",
+    "xor_eq",
+    "compl",
+    // Metal-specific
+    "constant",
+    "device",
+    "threadgroup",
+    "threadgroup_imageblock",
+    "compute",
```

Comment:
I wonder how we could easily find keywords like `kernel` to include here. We might also want to consider adding most of the C++ keywords which includes words like `using`

Reply:
Added in https://github.com/gfx-rs/naga/commit/6e42c3468306d410a7473397525a45da0f65e420